src/ice.js

```diff
@@ -227,6 +221,73 @@
 				});
 			},
 
+			getTrackedUserIds: function(trackedElements) {
+				var self = this;
+				return Array.prototype.map.call(trackedElements, function(elem) {
+					var value = elem.attributes[self.userIdAttribute];
+					return value ? parseInt(value.nodeValue, 10) : self._hashNameToUserId(elem);
+				}).filter(function(elem, idx, arr) {
+					return arr.indexOf(elem) == idx;
+				});
+			},
+
+			_retrieveOrGenerateStyleIndex: function(nodeEl, userid) {
+				if (this._trackedUsersId.indexOf(userid)) {
+					var matchStyleIndex = nodeEl.className.match(/\d+/); // retrieve from current class 'cts-1'
+					if (matchStyleIndex) {
+						return matchStyleIndex[0];
+					}
+				}
+
+				return this._trackedUsersId.length % this.styleColorsNumber + 1;
+			},
+
+			_retrieveOrGenerateUserId: function(nodeEl) {
+				var userid = ice.dom.attr(nodeEl, this.userIdAttribute);
+				return userid ? userid : this._hashNameToUserId(nodeEl); 
+			},
+
+			_hashNameToUserId: function(nodeEl) {
+				var name = ice.dom.attr(nodeEl, this.userNameAttribute);
+				var hash = 0, i, chr;
```

Comment:
i et chr non pas a être déclarées en dehors du scope de la boucle for

Reply:
done